include/zephyr/net/coap_client.h

```diff
@@ -159,6 +159,28 @@ int coap_client_req(struct coap_client *client, int sock, const struct sockaddr
  */
 void coap_client_cancel_requests(struct coap_client *client);
 
+/**
+ * @brief Populate a Block2 option to be added to a request
+ *
+ * If the application expects a request to require a blockwise transfer, it may pre-emptively
+ * suggest a maximum block size to the server - see RFC7959 Figure 3: Block-Wise GET with Early
+ * Negotiation.
+ *
+ * This helper function populates the Block2 option to send with the initial request.
+ *
+ * @param option CoAP client option structure to populate
+ */
+static inline void coap_client_option_populate_block2(struct coap_client_option *option)
+{
+	if (option == NULL) {
+		return;
+	}
```

Comment:
Second option would be that this function would just return the structure.
I think playing with the pointer is unnecessary

```suggestion
static inline struct coap_client_option coap_client_option_block2()
{
        return (struct coap_client_option) { .code = COAP_OPTION_BLOCK2, ...
```